semgrep/semgrep/rule.py

```diff
@@ -70,213 +52,61 @@ def __init__(self, raw: YamlTree[YamlMap]) -> None:
         # add typescript to languages if the rule supports javascript.
         if any(language == Language.JAVASCRIPT for language in rule_languages):
             rule_languages.add(Language.TYPESCRIPT)
+            self._raw["languages"] = [r.value for r in rule_languages]
 
         self._languages = sorted(rule_languages, key=lambda lang: lang.value)  # type: ignore
 
         # check taint/search mode
         if self._raw.get("mode") == JOIN_MODE:
             self._mode = JOIN_MODE
         else:
-            self._expression, self._mode = self._build_search_patterns_for_mode(
```

Comment:
We no longer need to parse the rule in the python side. Most changes to this file are deleting this code path